tools/tf_graph_tool.py

```diff
@@ -167,23 +169,121 @@ def print_graph_stat(input_path):
             logging.info("\t%s = %s", op, count)
 
     @staticmethod
-    def extract_sub_graph(input_path, output_path=None, dest_nodes=None):
-        if not output_path:
-            output_path = append_file_name_suffix(input_path, "sub")
-
+    def extract_sub_graph(input_path, dest_nodes=None, output_path=None,
+                          src_nodes=None, name_prefix=None):
         logging.info("load from %s", input_path)
         graph_def = load_graph_def_from_pb(input_path)
         logging.info("\ttotal node = %s", len(graph_def.node))
 
-        if dest_nodes:
-            dest_nodes = dest_nodes.split(',')
+        if (dest_nodes or src_nodes) and name_prefix:
+            raise RuntimeError("dest_nodes/src_nodes and name_prefix are incompatible.")
+        if not name_prefix:
+            name_prefix = ""
+            if not dest_nodes:
+                _, dest_nodes, _ = get_graph_def_io_nodes(graph_def)
         else:
-            _, dest_nodes = get_graph_def_io_nodes(graph_def)
+            dest_nodes = []
+            for node in graph_def.node:
+                if node.name.startswith(name_prefix):
+                    dest_nodes.append(node.name)
+        if not src_nodes:
+            src_nodes = []
+
+        if not isinstance(dest_nodes, list):
```

Comment:
utils.make_sure?

Reply:
don't want to use converter's utilities.